client/consensus/grandpa/lib.go

```
@@ -24,3 +26,37 @@ type NewAuthoritySetStruct[H comparable, N constraints.Unsigned] struct {
 	SetId       N
 	Authorities []Authority
 }
+
+type msgData[H comparable, N constraints.Unsigned] struct {
+	Round   uint64
+	SetID   uint64
+	Message finalityGrandpa.Message[H, N]
+}
+
+// Check a message signature by encoding the message as a localized payload and
+// verifying the provided signature using the expected authority id.
+// The encoding necessary to verify the signature will be done using the given
+// buffer, the original content of the buffer will be cleared.
+func checkMessageSignature[H comparable, N constraints.Unsigned](
```

this function should return an error since it might fail to marshal/verify